aspnetcore/signalr/redis-backplane.md

```diff
@@ -0,0 +1,187 @@
+  * [Nginx](https://docs.nginx.com/nginx/admin-guide/load-balancer/http-load-balancer/#sticky)
+  * [pfSense](https://www.netgate.com/docs/pfsense/loadbalancing/inbound-load-balancing.html#sticky-connections)
+
+## Redis server errors
+
+When a Redis server goes down, SignalR throws exceptions that indicate messages won't be delivered. Some typical exception messages:
+
+* *Failed writing message*
+* *Failed to invoke hub method 'MethodName'*
+* *Connection to Redis failed*
+
+SignalR doesn't buffer messages to send them when the server comes back up. Any messages sent while the Redis server is down are lost.
+
+SignalR automatically reconnects when the Redis server is available again.
+
+### Custom behavior for connection failures
+
+Here's an example that shows how to handle Redis connection failure events.
+
+::: moniker range="= aspnetcore-2.1"
+
+```csharp
+services.AddSignalR()
+        .AddRedis(o =>
+        {
+            o.ConnectionFactory = async writer =>
+            {
+                var config = new ConfigurationOptions
+                {
+                    AbortOnConnectFail = false
+                };
+                config.EndPoints.Add(IPAddress.Loopback, 0);
+                config.SetDefaultPorts();
+                var connection = await ConnectionMultiplexer.ConnectAsync(config, writer);
+                connection.ConnectionFailed += (_, e) =>
+                {
+                    Console.WriteLine("Connection to Redis failed.");
+                };
+
+                if (!connection.IsConnected)
+                {
+                    Console.WriteLine("Did not connect to Redis.");
+                }
+
+                return connection;
+            };
+        });
+```
+
+::: moniker-end
+
+::: moniker range="> aspnetcore-2.1"
+
+```csharp
+services.AddSignalR()
+        .AddMessagePackProtocol()
+        .AddStackExchangeRedis(o =>
+        {
+            o.ConnectionFactory = async writer =>
+            {
+                var config = new ConfigurationOptions
+                {
+                    AbortOnConnectFail = false
+                };
+                config.EndPoints.Add(IPAddress.Loopback, 0);
+                config.SetDefaultPorts();
+                var connection = await ConnectionMultiplexer.ConnectAsync(config, writer);
+                connection.ConnectionFailed += (_, e) =>
+                {
+                    Console.WriteLine("Connection to Redis failed.");
+                };
+
+                if (!connection.IsConnected)
+                {
+                    Console.WriteLine("Did not connect to Redis.");
+                }
+
+                return connection;
+            };
+        });
+```
+
+::: moniker-end
+
+## Clustering
+
+Clustering is a method for achieving high availability by using multiple Redis servers. Clustering is not officially supported, but it might work.
```

Comment:
```suggestion
Clustering is a method for achieving high availability by using multiple Redis servers. Clustering isn't officially supported, but it might work.
```